pkg/fission-cli/cmd/httptrigger/update.go

```diff
@@ -148,7 +149,14 @@ func (opts *UpdateSubCommand) complete(input cli.Input) (err error) {
 }
 
 func (opts *UpdateSubCommand) run(input cli.Input) error {
-
+	if input.Bool(flagkey.SpecSave) {
+		specFile := fmt.Sprintf("route-%v.yaml", opts.trigger.ObjectMeta.Name)
+		err := spec.SpecSaveUpdate(*opts.trigger, specFile)
```

Comment:
We should validate spec before calling SpecSaveUpdate function, same as updating environment.

Reply:
Added validation for `Function`, `HTTP trigger,` `MQ trigger`, `Time trigger` update too.